packages/amplify-graphql-conversation-transformer/src/transformer-steps/conversation-resolver-generator.ts

```diff
@@ -109,7 +110,18 @@ export class ConversationResolverGenerator {
     functionStack: cdk.Stack,
     capitalizedFieldName: string,
   ): { functionDataSourceId: string; referencedFunction: IFunction } {
-    if (directive.functionName) {
+    if (directive.handler && directive.functionName) {
+      throw new Error("'functionName' and 'handler' are mutually exclusive");
+    }
+    if (directive.handler) {
+      const eventVersion = semver.coerce(directive.handler.eventVersion);
+      if (eventVersion?.major !== 1) {
+        throw new Error(
+          `Unsupported custom conversation handler. Expected eventVersion to match 1.x, received ${directive.handler.eventVersion}`,
+        );
+      }
```

Comment:
This seems more appropriate to house in the transformer's `validate` method.

Reply:
I had a feeling this is too deep in the implementation details, but couldn't find better place.
I'll take a stab at `validate` method, thanks for the pointer.

Reply:
done.